client/subcomponents/produce/ProduceList/ProduceList.jsx

```diff
@@ -1,17 +1,24 @@
 import React, { useEffect, useState } from 'react'
 import { getProduce } from './produceHelper'
+import { showError } from '../../../slices/error'
+import { useDispatch } from 'react-redux'
 
 export default function ProduceList({ gardenid }) {
+  const dispatch = useDispatch()
   const [produce, setProduce] = useState([])
 
   useEffect(() => {
-    // eslint-disable-next-line promise/catch-or-return
-    getProduce(gardenid).then(({ produce }) => {
-      setProduce(produce)
-      return null
-    })
+    const retrieveProduce = async () => {
+      const gardenProducts = await getProduce(gardenid)
+      await setProduce(gardenProducts.produce)
+    }
+    retrieveProduce()
+      .catch((error) => {
```

Comment:
.catch() is .then() syntax - can you please use a try catch block here instead?

Reply:
Thank you!